src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,83 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import React, { ChangeEvent, useState } from 'react';
+
+import { Filter } from '../../App';
+
+interface Props {
+  onFilter: (filters: Filter) => void;
+}
+
+export const TodoFilter: React.FC<Props> = ({ onFilter }) => {
+  const [statusFilter, setStatusFilter] = useState('all');
+  const [titleFilter, setTitleFilter] = useState('');
+
+  const handleChangeStatus = (event: ChangeEvent<HTMLSelectElement>) => {
+    setStatusFilter(event.target.value);
+    onFilter({ status: event.target.value, title: titleFilter });
+  };
+
+  const handleChangeInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitleFilter(event.target.value);
+    onFilter({
+      status: statusFilter,
+      title: event.target.value.toLocaleLowerCase().trim(),
+    });
+  };
+
+  const handleClear = () => {
+    setStatusFilter('all');
+    setTitleFilter('');
+    onFilter({ status: 'all', title: '' });
+  };
+
+  const statusOptions = [
+    { value: 'all', lable: 'All' },
+    { value: 'active', lable: 'Active' },
+    { value: 'completed', lable: 'Completed' },
```

Comment:
There is a typo in the `statusOptions` array: 'lable' should be 'label'. This typo will affect the display of the dropdown options.